crates/core/tedge_api/src/workflow/supervisor.rs

```diff
@@ -229,6 +298,152 @@ impl WorkflowSupervisor {
     }
 }
 
+/// A stack of known versions for a workflow
+///
+/// In practice, one might have 3 concurrent versions:
+///
+/// - The current version, i.e. the version to be used for new operation instances.
+/// - One in-use version, that was used by some operation instance when the current version has been updated.
+/// - The builtin version provided by the agent and which can be overridden by the users.
+///
+/// None of these versions are mandatory. For instance, an operation can have no builtin version,
+/// no current version (because the user removed the definition file), but a version still in-use
+/// (that has been started before the user deprecated the operation).
+struct WorkflowVersions {
+    operation: OperationName,
+    current: Option<WorkflowVersion>,
```

Comment:
While reviewing the code, I was often getting `current` and `in-use` confused and mixed up, esp with `current` as it could be wrongly interpreted as the "current one being used". Calling it something like `latest` might help.

Reply:
Unfortunately, `latest` is used for the on-disk version.

- `current` is the version currently loaded by the workflow engine.
- `in-use` is a copy that is kept for a running workflow in case the agent restarts while the operation is still under execution. On restart that version must be used and not a new one.